aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -912,7 +913,7 @@ Tensor mm_batching_rule(const Tensor& self, const Tensor& other) {
   TORCH_INTERNAL_ASSERT(false, "either self or other must be a BatchedTensor");
 }
 
-Tensor cat_batching_rule(TensorList tensors, int64_t dim) {
+Tensor cat_batching_rule(const ITensorListRef& tensors, int64_t dim) {
```

Comment:
Is there a way we decide between passing const reference vs. passing by value? (I don't remember if the ABI says they'll end up the same)

Reply:
Whenever it's possible, I'm passing by value. But, registering to the dispatcher [requires const-ref](https://github.com/pytorch/pytorch/issues/66328#issuecomment-998100548) due to include dependencies (at least, it was needed at the time I implemented this).